packages/smooth_app/lib/pages/product/common/product_list_page.dart

```diff
@@ -48,75 +49,159 @@ class _ProductListPageState extends State<ProductListPage> {
+                    )) {
+                      localDatabase.notifyListeners();
+                    }
+                    break;
+                  default:
+                    throw Exception('Unknown value: $value');
+                }
               },
             ),
+        ],
+      ),
       body: products.isEmpty
           ? Center(
               child: Text(appLocalizations.no_prodcut_in_list,
                   style: Theme.of(context).textTheme.subtitle1),
             )
           : ListView.builder(
-              itemCount: products.length,
+              itemCount: products.length + 1,
               itemBuilder: (BuildContext context, int index) {
+                if (index == 0) {
+                  // header
+                  return Row(
+                    mainAxisAlignment: MainAxisAlignment.spaceAround,
+                    children: <Widget>[
+                      if (_selectionMode && _selectedBarcodes.isNotEmpty)
+                        ElevatedButton(
+                          child: const Text(
+                              'Compare'), // TODO(monsieurtanuki): localize
+                          onPressed: () async {
+                            final List<Product> list = <Product>[];
+                            for (final Product product in products) {
+                              if (_selectedBarcodes.contains(product.barcode)) {
+                                list.add(product);
+                              }
+                            }
+                            await Navigator.push<Widget>(
+                              context,
+                              MaterialPageRoute<Widget>(
+                                builder: (BuildContext context) =>
+                                    PersonalizedRankingPage.fromItems(
+                                  products: list,
+                                  title:
+                                      '', // TODO(monsieurtanuki): find inspiration
+                                ),
+                              ),
+                            );
+                            setState(() => _selectionMode = false);
+                          },
+                        ),
+                      if (_selectionMode)
+                        ElevatedButton(
+                          onPressed: () =>
+                              setState(() => _selectionMode = false),
+                          child: Text(AppLocalizations.of(context)!.cancel),
+                        ),
+                      if (!_selectionMode)
+                        ElevatedButton(
+                          onPressed: () => setState(
+                            () {
+                              _selectionMode = true;
+                              _populateAll(products);
+                            },
+                          ),
+                          child: const Text(
+                              'Compare Mode'), // TODO(monsieurtanuki): localize
```

Comment:
optional: why not just localize in this PR :D

Reply:
Good idea, but that would mean the labels are not supposed to change.
And I don't think it's the case as we're still discussing the UI/UX, e.g. if we finally put `'14/16 products'`, and where.